etna/packages/magmaR/R/retrieveMatrix.R

```diff
@@ -91,16 +91,26 @@ retrieveMatrix <- function(
         token = token,
         ...)
     
-    # Turn into matrix
+    # Extract matrix data as a list of columns
     data_cols <- lapply(
         recordNames,
         function(x) {
             (json$models[[modelName]]$documents[[x]])[[attributeNames]]
         })
-    data <- as.matrix(do.call(data.frame, data_cols))
+    
+    # Identify any empty records
+    empty <- vapply(data_cols, FUN = length, FUN.VALUE = integer(1)) == 0
```

Comment:
Just curious, is there ever a case where you would want the empty entries there for analysis purposes (like to know you are missing data), and so this filtering would be helpful behind a flag, with a default value? Or basically empty entries are never really useful, so baking in this data-cleaning step is super useful?

Reply:
Hmmm.  I could definitely see returning the ids of empty records altogether being useful.  I think the way these records come into existence is likely either 1) data gets added for other attributes of the model, but not the target one being retrieved, or 2) a record is "removed".  So perhaps useful for QC?

In this function, the trimming is needed in order to have a matrix as output -- length 0 entries break the rectangular shape of the potential matrix -- and I think we want to keep this output structure.

I do report identifiers of all these records in a warning a few lines below.

But it wouldn't be super easy to scrape the full set of ids together from the warnings... Perhaps collection of such records could be a separate endpoint added in a future PR?

Reply:
Oh, I see, so you have to keep the shape. Yeah, certainly don't worry about it for this PR, maybe if someone needs it they can also use lower-level functions (using this as an example), in their own R code. Or as we collect feedback, we can see if building in such a feature would be useful or not to bake in.